pkg/resources/kafka/kafka.go

```diff
@@ -905,6 +911,30 @@ func (r *Reconciler) createExternalListenerStatuses(log logr.Logger) (map[string
 	return extListenerStatuses, nil
 }
 
+func (r *Reconciler) getK8sAssignedNodeport(log logr.Logger, eListenerName string, brokerId int32) (int32, error) {
+	log.Info("determining automatically assigned nodeport",
+		"brokerId", brokerId, "listenerName", eListenerName)
+	nodePortSvc := &corev1.Service{}
+	err := r.Client.Get(context.Background(),
+		types.NamespacedName{Name: fmt.Sprintf("%s-%d-%s", r.KafkaCluster.GetName(), brokerId, eListenerName),
```

Comment:
can we factor out the nodeport service name generation logic into a separate function and use it here and also where the service is created?